src/main.rs

```diff
@@ -565,42 +568,71 @@ fn run_cli(cli: Cli) -> Result<()> {
             };
             let remote_id = RemoteId::local(&alias);
 
-            let (row, data, canonical) = match record_db.state_from_remote_id(&remote_id)? {
-                RemoteIdState::Existent(row) => {
-                    if from.is_some() {
-                        row.commit()?;
-                        bail!("Local record '{remote_id}' already exists")
-                    } else {
+            let (row, data) = if let Some(old_id) = move_from {
+                // Allowing for arbitrary `old_id` without validation and trimming
+                // so that local ids that were valid in an older version can be renamed.
+                // SAFETY: This is safe as a colon is present in the `full_id`.
+                let old_remote_id =
+                    unsafe { RemoteId::from_string_unchecked("local:".to_owned() + &old_id) };
+                match record_db.state_from_remote_id(&old_remote_id)? {
+                    RemoteIdState::Existent(row) => {
+                        if !row.change_row_canonical_id(&remote_id)? {
+                            bail!("Local record '{remote_id}' already exists")
+                        }
+
+                        if let Ok(old_alias) = Alias::from_str(&old_id) {
+                            row.check_and_delete_alias(&old_alias)?;
+                        }
+
                         let raw_record_data = row.get_data()?.data;
-                        (row, raw_record_data, remote_id)
+                        (row, raw_record_data)
+                    }
+                    RemoteIdState::Null(null_row) => {
+                        null_row.commit()?;
+                        bail!("'{remote_id}' was found in the 'NullRecords' table. A local record should not be present in the 'NullRecords' table.");
```

Comment:
Could suggest fixing with `autobib delete --delete-null`

Reply:
I thought about this but the `suggest!` would have to come before the `bail!` and that didn't feel quite right. Not sure what's best

Reply:
Hmm, good point, that's a bit awkward. Because of the way errors are handled a hacky (but equivalent) solution is to just use `error!` and then `suggest!` and then` return Ok(())`

Reply:
But agreed this feels non-ergonomic.

Reply:
Alright, I've implemented the hack-y solution.